scripts/release/publish.js

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env node
+
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+const execa = require('execa');
+
+const ARGS = [
+    // https://github.com/lerna/lerna/tree/master/commands/publish
+    'publish',
+    // Explicitly publish packages tagged in the current commit
+    'from-git',
+    // Disable lifecycle scripts, specifically `prepare`
+    '--ignore-scripts',
+    // Lerna will otherwise attempt to verify npm user access
+    '--no-verify-access',
+    '--yes',
+];
+
+const { stderr, stdin, stdout } = process;
+
+const RELEASE_BRANCH_RE = /^(master|winter\d\d|spring\d\d|summer\d\d)$/;
+
+try {
+    if (execa.sync('echo', ['$CI'], { shell: true }).stdout !== 'true') {
```

Comment:
We already have `is-ci` dependency covering this. It is used in the build process to limit the numbers of cores.